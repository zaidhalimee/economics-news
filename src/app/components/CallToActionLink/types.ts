import { EventTrackingMetadata } from '#app/models/types/eventTracking';
import { GelFontSize, FontVariant } from '../../models/types/theming';

export type CallToActionLinkProps = {
  href?: string;
  className?: string;
  eventTrackingData?: EventTrackingMetadata;
  download?: boolean;
  size?: GelFontSize;
  fontVariant?: FontVariant;
  ignoreLiteExtension?: boolean;
  centerAlign?: boolean;
  text?: string;
};
