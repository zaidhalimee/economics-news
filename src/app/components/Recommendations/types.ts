export type Recommendation = {
  title: string;
  image: {
    width: number;
    height: number;
    altText: string;
    copyrightHolder: string;
    originCode: string;
    locator: string;
  };
  href: string;
};
