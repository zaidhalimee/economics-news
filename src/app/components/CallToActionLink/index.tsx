/** @jsx jsx */
/* @jsxFrag React.Fragment */
import React, { PropsWithChildren } from 'react';
import { jsx } from '@emotion/react';
import useClickTrackerHandler from '#app/hooks/useClickTrackerHandler';
import Text from '../Text';
import styles from './index.styles';
import { CallToActionLinkProps } from './types';

const CallToActionLink = ({
  href,
  className,
  children,
  eventTrackingData,
  size = 'pica',
  fontVariant = 'sansBold',
  download = false,
  ignoreLiteExtension = false,
  linkText = '',
  borderStyleOverride = false,
}: PropsWithChildren<CallToActionLinkProps>) => {
  const clickTrackerHandler = useClickTrackerHandler(eventTrackingData);

  return (
    <a
      href={href}
      className={className}
      css={styles.linkBackground}
      onClick={clickTrackerHandler}
      download={download}
      {...(ignoreLiteExtension && { 'data-ignore-lite': true })}
    >
      {borderStyleOverride ? (
        <>
          <Text size={size} fontVariant={fontVariant}>
            {linkText}
          </Text>
          {children}
        </>
      ) : (
        <div css={styles.linkTextWrapper}>
          <Text size={size} fontVariant={fontVariant} css={styles.linkText}>
            {children}
          </Text>
        </div>
      )}
    </a>
  );
};

export default CallToActionLink;
