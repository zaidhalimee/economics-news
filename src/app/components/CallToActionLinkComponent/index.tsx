/** @jsx jsx */
import { PropsWithChildren } from 'react';
import { jsx } from '@emotion/react';
import useClickTrackerHandler from '#app/hooks/useClickTrackerHandler';
import { EventTrackingMetadata } from '#app/models/types/eventTracking';
import Chevron from './Chevron';
import FlexWrapper from './FlexWrapper';

type CallToActionLinkProps = {
  to?: string;
  className?: string;
  eventTrackingData?: EventTrackingMetadata;
};

const CallToActionLink = ({
  to,
  children,
  eventTrackingData,
  ...htmlAttributes // handles LiteSiteCta
}: PropsWithChildren<CallToActionLinkProps>) => {
  const clickTrackerHandler = useClickTrackerHandler(eventTrackingData);
  const onClick = eventTrackingData ? clickTrackerHandler : () => null;

  return (
    <a href={to} onClick={onClick} {...htmlAttributes}>
      {children}
    </a>
  );
};

CallToActionLink.Chevron = Chevron;
CallToActionLink.FlexWrapper = FlexWrapper;

export default CallToActionLink;
