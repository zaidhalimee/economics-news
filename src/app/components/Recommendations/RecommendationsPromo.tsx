/** @jsx jsx */
import { jsx } from '@emotion/react';
import { useContext } from 'react';

import { RequestContext } from '#contexts/RequestContext';
// import RecommendationsImage from '../RecommendationsPromoImage';
// import useCombinedClickTrackerHandler from '../../StoryPromo/useCombinedClickTrackerHandler';
// import extractPromoData from './RecommendationsPromo/utility';
import styles from './index.styles';

type Props = {
  eventTrackingData?: {
    componentName: string;
  } | null;
};

const RecommendationsPromo = ({ promo, eventTrackingData = null }: Props) => {
  const { isLite } = useContext(RequestContext);
  const handleClickTracking = useCombinedClickTrackerHandler(eventTrackingData);

  // const { headline, url, indexImage } = extractPromoData({ promo });

  return (
    <div css={styles.promoWrapper} data-e2e="story-promo-wrapper">
      {!isLite && (
        <div css={styles.imageWrapper}>
          <RecommendationsImage indexImage={indexImage} lazyLoad />
        </div>
      )}
      <div css={styles.textWrapper}>
        <div css={styles.headline}>
          <a
            css={styles.link}
            href={url}
            onClick={eventTrackingData ? handleClickTracking : null}
          >
            {headline}
          </a>
        </div>
      </div>
    </div>
  );
};

export default RecommendationsPromo;
