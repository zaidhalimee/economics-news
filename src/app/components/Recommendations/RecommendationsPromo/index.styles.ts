import { css, Theme } from '@emotion/react';

export default {
  promoWrapper: ({ palette, spacings }: Theme) =>
    css({
      display: 'flex',
      alignItems: 'center',
      position: 'relative',
      padding: `${spacings.FULL}rem`,
      backgroundColor: palette.GHOST,
    }),
  imageWrapper: ({ mq }: Theme) =>
    css({
      display: 'inline-block',
      position: 'relative',
      width: '4.7rem',
      verticalAlign: 'top',

      [mq.GROUP_2_MIN_WIDTH]: {
        width: '6.8rem',
      },

      [mq.GROUP_3_MIN_WIDTH]: {
        width: '7.5rem',
      },
    }),
  textWrapper: ({ spacings, mq, isLite }: Theme) =>
    css({
      display: 'inline-block',
      width: isLite ? '100%' : 'calc(100% - 7.5rem)',
      padding: `0 ${spacings.FULL}rem`,
      verticalAlign: 'top',
      height: '100%',

      [mq.GROUP_1_MAX_WIDTH]: {
        width: 'calc(100% - 5rem)',
      },

      [mq.GROUP_3_MIN_WIDTH]: {
        padding: `0 ${spacings.DOUBLE}rem`,
      },
    }),
  link: ({ palette }: Theme) =>
    css({
      position: 'static',
      color: palette.GREY_10,
      textDecoration: 'none',
      overflowWrap: 'break-word',

      '&:before': {
        bottom: 0,
        content: "''",
        left: 0,
        overflow: 'hidden',
        position: 'absolute',
        right: 0,
        top: 0,
        whiteSpace: 'nowrap',
        zIndex: 1,
      },

      '&:hover, &:focus': {
        textDecoration: 'underline',
      },

      '&:visited': {
        color: palette.METAL,
      },
    }),
  headline: ({ palette, fontSizes, fontVariants }: Theme) =>
    css({
      ...fontVariants.serifMedium,
      ...fontSizes.pica,
      color: palette.GREY_10,
      margin: 0,
      height: '100%',
      display: 'flex',
      alignItems: 'center',
    }),
};
