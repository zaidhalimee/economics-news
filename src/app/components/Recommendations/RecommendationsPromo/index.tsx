/** @jsx jsx */
import { jsx } from '@emotion/react';
import { useContext } from 'react';

import { RequestContext } from '#contexts/RequestContext';
import useClickTrackerHandler from '#app/hooks/useClickTrackerHandler';
import RecommendationsImage from '../RecommendationsPromoImage';
import styles from './index.styles';
import { Recommendation } from '../types';

const eventTrackingData = {
  componentName: 'recommendations',
};

const RecommendationsPromo = ({
  recommendation,
}: {
  recommendation: Recommendation;
}) => {
  const { isLite } = useContext(RequestContext);
  const handleClickTracking = useClickTrackerHandler(eventTrackingData);

  if (!recommendation) return null;

  const { title, image, href } = recommendation;

  return (
    <div css={styles.promoWrapper} data-e2e="story-promo-wrapper">
      {!isLite && (
        <div css={styles.imageWrapper}>
          <RecommendationsImage image={image} lazyLoad />
        </div>
      )}
      <div css={styles.textWrapper}>
        <div css={styles.headline}>
          <a
            css={styles.link}
            href={href}
            {...(eventTrackingData && { onClick: handleClickTracking })}
          >
            {title}
          </a>
        </div>
      </div>
    </div>
  );
};

export default RecommendationsPromo;
