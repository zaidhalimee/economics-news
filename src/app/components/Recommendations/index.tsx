/** @jsx jsx */
import { useContext } from 'react';
import { jsx, useTheme } from '@emotion/react';

import SectionLabel from '#psammead/psammead-section-label/src';
import SkipLinkWrapper from '#components/SkipLinkWrapper';
import useToggle from '#hooks/useToggle';

import { ServiceContext } from '#contexts/ServiceContext';
// import RecommendationsPromoList from './RecommendationsPromoList';
import RecommendationsPromo from './RecommendationsPromo';
import styles from './index.styles';
import { Recommendation } from './types';

const Recommendations = ({ data }: { data: Recommendation[] }) => {
  const { recommendations, translations, script, service, dir } =
    useContext(ServiceContext);
  const { enabled } = useToggle('cpsRecommendations');

  const {
    palette: { GREY_2 },
  } = useTheme();

  const labelId = 'recommendations-heading';
  const a11yAttributes = {
    as: 'section',
    role: 'region',
    'aria-labelledby': labelId,
  };

  const { hasStoryRecommendations } = recommendations || {};
  console.log(data);
  // if (!hasStoryRecommendations || !enabled || !data.length) return null;

  const title = translations?.recommendationTitle ?? 'Recommended stories';

  const {
    skipLink: { text, endTextVisuallyHidden },
  } = recommendations;

  const terms = { '%title%': title };

  const isSinglePromo = data?.length === 1;

  const endTextId = `end-of-recommendations`;

  const skipLink = {
    endTextId,
    terms,
    text,
    endTextVisuallyHidden,
  };

  return (
    <div
      css={styles.recommendationsWrapper}
      data-e2e={labelId}
      {...a11yAttributes}
    >
      <SkipLinkWrapper service={service} {...skipLink}>
        {title ? (
          <SectionLabel
            css={styles.labelComponent}
            script={script}
            service={service}
            dir={dir}
            labelId={labelId}
            columnType="main"
            mobileDivider={false}
            overrideHeadingAs="strong"
            bar={false}
            backgroundColor={GREY_2}
          >
            {title}
          </SectionLabel>
        ) : null}
        <RecommendationsPromo recommendation={data?.[0]} />
        {/* {isSinglePromo ? (
              <RecommendationsPromo promo={data[0]} />
            ) : (
              <RecommendationsPromoList promoItems={data} />
            )} */}
      </SkipLinkWrapper>
    </div>
  );
};

export default Recommendations;
