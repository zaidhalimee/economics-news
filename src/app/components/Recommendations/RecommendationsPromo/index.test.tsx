import React from 'react';
import { ToggleContextProvider } from '#contexts/ToggleContext';
import { render } from '#app/components/react-testing-library-with-providers';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import RecommendationsPromo from '.';
import { Recommendation } from '../types';

const recommendationFixture = {
  title: 'शेख़ मुजीब और इंदिरा गांधी के बीच हुए समझौते पर क्यों हुआ था विवाद?',
  image: {
    width: 1824,
    height: 1026,
    altText: 'सुनीता विलियम्स',
    locator: 'a095/live/9c11d5e0-0581-11f0-88b7-5556e7b55c5e.jpg',
    originCode: 'cpsprodpb',
    copyrightHolder: 'Getty Images',
  },
  href: 'https://www.bbc.co.uk',
};

const Component = ({ recommendation }: { recommendation: Recommendation }) => {
  return (
    <ServiceContextProvider service="pidgin">
      <ToggleContextProvider
        toggles={{
          eventTracking: {
            enabled: true,
          },
        }}
      >
        <RecommendationsPromo recommendation={recommendation} />,
      </ToggleContextProvider>
    </ServiceContextProvider>
  );
};

describe('RecommendationsPromo', () => {
  it('it renders a Recommendations Promo', () => {
    const { container } = render(
      <RecommendationsPromo recommendation={recommendationFixture} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render the title of the article as a link', () => {
    const { getByText, container } = render(
      <Component recommendation={recommendationFixture} />,
    );

    const links = container.querySelectorAll('a');

    expect(getByText(recommendationFixture.title)).toBeInTheDocument();
    expect(links).toHaveLength(1);
    expect(links[0].getAttribute('href')).toEqual(recommendationFixture.href);
  });
});
