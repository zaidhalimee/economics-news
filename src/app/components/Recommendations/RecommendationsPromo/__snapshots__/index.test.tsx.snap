// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecommendationsPromo it renders a Story Promo wrapped in a Grid component 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  padding: 0.5rem;
  background-color: #FDFDFD;
}

.emotion-1 {
  display: inline-block;
  position: relative;
  width: 4.7rem;
  vertical-align: top;
}

@media (min-width: 25rem) {
  .emotion-1 {
    width: 6.8rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-1 {
    width: 7.5rem;
  }
}

.emotion-2 {
  position: relative;
  height: 0;
  background-image: url(data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48c3ZnIHdpZHRoPSI0NzkiIGhlaWdodD0iMTM2IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPjxnIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCI+PGcgZmlsbD0iI0Q1RDBDRCI+PGc+PHBhdGggZD0iTTQ0OS41NTYgOTEuNTUzbC0yLjMxIDEuNDFjLTExLjE2NyA2LjgxOC0yMy4zMTMgMTAuNDc0LTM0LjM0NiAxMC40Ny0yMy42MS0uMDktMzkuMTYzLTE0LjA4My0zOS4yMjctMzQuNDUzLjAzLTE5LjkxOCAxNi4yNy0zNC42NjMgMzguNjMzLTM0LjcxOCAxMS4zODcuMDEgMjEuNzAzIDIuOTU0IDMzLjk2MiA5LjY3MmwyLjI1MSAxLjI0di0xOC4xOWwtLjk2Mi0uMzc3Yy0xMy44MjQtNS40NTgtMjQuNTMtNy44OS0zNS4xMDMtNy44ODQtMTYuMzQ2LS4wMDYtMzAuNTMzIDUuMzk0LTQwLjYzNyAxNC41NTctMTAuMTA1IDkuMTYzLTE2LjEwNiAyMi4xMDItMTYuMDk5IDM2Ljk1My4wMDggMTAuMzQ4IDQuMjc5IDIyLjQ4IDEzLjQyIDMyLjEwNSA5LjEyMSA5LjYyOCAyMy4xNjUgMTYuNjQ4IDQyLjQzIDE2LjYzOWguMDYzYzE1Ljk4IDAgMjcuMDYyLTMuNTYzIDM3LjA3NC04LjQ5MmwuODUxLS40MTRWOTEuNTUzek0zMzQgMTM1LjY5N2gxNDQuMTk1VjBIMzM0djEzNS42OTd6Ii8+PHBhdGggZD0iTTI3Ni45MzcgODkuOTY4Yy4wNDEtMTIuMzMtOC4xNzEtMjEuNjk2LTIxLjMwOC0yNS4zIDMuNTQ0LTEuODA5IDYuMzUtNC4wMjMgOC40MDQtNi43MjcgMi43NS0zLjYyMiA0LjA2MS04LjA2NCA0LjA0Ni0xMy4yMzUuMDE1LTYuMzU5LTIuNDg2LTEyLjgzOS03Ljg1OC0xNy42ODctNS4zNzItNC44NDctMTMuNTI2LTcuOTk3LTI0LjY1NC03Ljk5MUgyMDQuODN2OTcuNzI4aDM2LjA3M2MxMi44NyAwIDIxLjkwNi0zLjQ4MiAyNy43MjItOC42NSA1LjgxOC01LjE1NSA4LjMyLTExLjkxIDguMzEyLTE4LjEzOHpNMTY3IDEzNS42OThoMTQ0LjE5N1YwSDE2N3YxMzUuNjk3eiIvPjxwYXRoIGQ9Ik0xMDkuOTM3IDg5Ljk2OGMuMDQxLTEyLjMzLTguMTcxLTIxLjY5Ni0yMS4zMDgtMjUuMyAzLjU0NC0xLjgwOSA2LjM1LTQuMDIzIDguNDA0LTYuNzI3IDIuNzUtMy42MjIgNC4wNjEtOC4wNjQgNC4wNDYtMTMuMjM1LjAxNS02LjM1OS0yLjQ4Ni0xMi44MzktNy44NTgtMTcuNjg3LTUuMzcyLTQuODQ3LTEzLjUyNi03Ljk5Ny0yNC42NTQtNy45OTFIMzcuODN2OTcuNzI4aDM2LjA3M2MxMi44NyAwIDIxLjkwNi0zLjQ4MiAyNy43MjItOC42NSA1LjgxOC01LjE1NSA4LjMyLTExLjkxIDguMzEyLTE4LjEzOHpNMCAxMzUuNjk4aDE0NC4xOTdWMEgwdjEzNS42OTd6Ii8+PHBhdGggZD0iTTI1OC42NjIgODguMTk4Yy0uMDEzIDMuMjI5LTEuMDA3IDYuNDc1LTMuODk2IDkuMDExLTIuODg0IDIuNTM3LTcuODczIDQuNDYzLTE2LjEzMyA0LjQ2M0gyMjJWNzVoMTUuODkzYzcuNDExIDAgMTIuNjcgMS41MDIgMTUuOTY1IDMuODUgMy4yODkgMi4zNjIgNC43NzYgNS40NjMgNC44MDQgOS4zNDgiLz48cGF0aCBkPSJNOTEuNjYyIDg4LjE5OGMtLjAxMyAzLjIyOS0xLjAwNyA2LjQ3NS0zLjg5NiA5LjAxMS0yLjg4NCAyLjUzNy03Ljg3NCA0LjQ2My0xNi4xMzMgNC40NjNINTVWNzVoMTUuODkyYzcuNDEyIDAgMTIuNjcyIDEuNTAyIDE1Ljk2NiAzLjg1IDMuMjg5IDIuMzYyIDQuNzc2IDUuNDYzIDQuODA0IDkuMzQ4Ii8+PHBhdGggZD0iTTI0NS4xODYgNTUuNzljMy4wOTYtMi4yMzcgNC41OS01LjM4NiA0LjYxMy0xMC4xMjQtLjAxNS0zLjI1LS45NDMtNi4wMzMtMy4yODEtOC4xMTEtMi4zNDYtMi4wNzgtNi4zMy0zLjU1NS0xMi43NTQtMy41NTVIMjIydjI1LjI3NWg4LjA3NmM2Ljk4OC4wMDQgMTEuOTk4LTEuMjQzIDE1LjExLTMuNDg2Ii8+PHBhdGggZD0iTTc4LjE4NiA1NS43OWMzLjA5Ni0yLjIzNyA0LjU5LTUuMzg2IDQuNjEzLTEwLjEyNC0uMDE1LTMuMjUtLjk0My02LjAzMy0zLjI4Mi04LjExMUM3Ny4xNzIgMzUuNDc3IDczLjE4OCAzNCA2Ni43NjQgMzRINTV2MjUuMjc1aDguMDc2YzYuOTg4LjAwNCAxMS45OTgtMS4yNDMgMTUuMTEtMy40ODYiLz48L2c+PC9nPjwvZz48L3N2Zz4K);
  -webkit-background-position: center center;
  background-position: center center;
  background-repeat: no-repeat;
  -webkit-background-size: 60px 17px;
  background-size: 60px 17px;
  background-color: #F2F2F2;
}

@media (min-width: 25rem) {
  .emotion-2 {
    -webkit-background-size: 77px 22px;
    background-size: 77px 22px;
  }
}

@media (min-width: 63rem) {
  .emotion-2 {
    -webkit-background-size: 93px 27px;
    background-size: 93px 27px;
  }
}

.emotion-3 {
  width: 100%;
  object-fit: cover;
  height: auto;
}

.emotion-4 {
  display: inline-block;
  width: calc(100% - 7.5rem);
  padding: 0 0.5rem;
  vertical-align: top;
  height: 100%;
}

@media (max-width: 24.9375rem) {
  .emotion-4 {
    width: calc(100% - 5rem);
  }
}

@media (min-width: 37.5rem) {
  .emotion-4 {
    padding: 0 1rem;
  }
}

.emotion-5 {
  font-family: Helmet,Freesans,Helvetica,Arial,sans-serif;
  font-style: normal;
  font-weight: 700;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #141414;
  margin: 0;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-5 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-5 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

.emotion-6 {
  position: static;
  color: #141414;
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow-wrap: break-word;
}

.emotion-6:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.emotion-6:hover,
.emotion-6:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-6:visited {
  color: #6E6E73;
}

<div>
  <div
    class="emotion-0"
    data-e2e="story-promo-wrapper"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
        style="padding-bottom: 56.25%; overflow: hidden;"
      >
        <img
          alt="सुनीता विलियम्स"
          class="emotion-3"
          loading="lazy"
          src="https://ichef.bbci.co.uk/ace/ws/660/cpsprodpb/a095/live/9c11d5e0-0581-11f0-88b7-5556e7b55c5e.jpg.webp"
          srcset="https://ichef.bbci.co.uk/ace/ws/70/cpsprodpb/a095/live/9c11d5e0-0581-11f0-88b7-5556e7b55c5e.jpg.webp 70w, https://ichef.bbci.co.uk/ace/ws/95/cpsprodpb/a095/live/9c11d5e0-0581-11f0-88b7-5556e7b55c5e.jpg.webp 95w, https://ichef.bbci.co.uk/ace/ws/144/cpsprodpb/a095/live/9c11d5e0-0581-11f0-88b7-5556e7b55c5e.jpg.webp 144w, https://ichef.bbci.co.uk/ace/ws/183/cpsprodpb/a095/live/9c11d5e0-0581-11f0-88b7-5556e7b55c5e.jpg.webp 183w, https://ichef.bbci.co.uk/ace/ws/240/cpsprodpb/a095/live/9c11d5e0-0581-11f0-88b7-5556e7b55c5e.jpg.webp 240w, https://ichef.bbci.co.uk/ace/ws/320/cpsprodpb/a095/live/9c11d5e0-0581-11f0-88b7-5556e7b55c5e.jpg.webp 320w, https://ichef.bbci.co.uk/ace/ws/660/cpsprodpb/a095/live/9c11d5e0-0581-11f0-88b7-5556e7b55c5e.jpg.webp 660w"
        />
      </div>
    </div>
    <div
      class="emotion-4"
    >
      <div
        class="emotion-5"
      >
        <a
          class="emotion-6"
          href="https://www.bbc.co.uk"
        >
          शेख़ मुजीब और इंदिरा गांधी के बीच हुए समझौते पर क्यों हुआ था विवाद?
        </a>
      </div>
    </div>
  </div>
</div>
`;
